### PR TITLE
Add #[error_type] attribute

### PR DESCRIPTION
**Pull request by Woyten**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use rocket_extra_codegen::FromRequest;
+
+#[derive(FromRequest)]
+#[error_type = "MyError"]
+struct BookingService {
+    db_conn: DbConnection,
+    queue_conn: QueueConnection,
+    user: AuthUser,
+}
+```
```
Comment (torfmaster) [inline README.md:28]: I would consider adding copyright/license notes in the Readme.

File: src/lib.rs [hunk 2]
```diff
@@ -49,13 +69,49 @@ fn try_derive_from_request(ast: &syn::DeriveInput) -> TokenStream {
...
     };
-    trait_implementation.into()
+    Ok(trait_implementation.into())
+}
+
+fn get_error_type(attrs: &[Attribute]) -> Result<Option<Type>, Error> {
+    let mut error_type_decls = attrs
+        .iter()
+        .filter_map(|attr| attr.parse_meta().ok().map(|meta| (attr, meta)))
+        .filter(|(_attr, meta)| meta.name() == "error_type")
```
Comment (torfmaster) [inline src/lib.rs]: Is there a way to handle the "error type" more consistently as string, attribute name?

File: src/lib.rs [hunk 3]
```diff
@@ -1,45 +1,65 @@
 #![recursion_limit = "128"]
+
```
Comment (torfmaster) [inline src/lib.rs:2]: Regarding formatting, should we fix a rust-toolchain?

**Review by torfmaster — APPROVED** [review]
LGTM. However, I have some questions of minor importance.

File: src/lib.rs (hunk 3 above)
Comment (Woyten) [reply src/lib.rs:2]: The formatting is stable since `rustfmt 1.0`. I added this line manually

File: src/lib.rs (hunk 2 above)
Comment (Woyten) [reply src/lib.rs]: Can you elaborate on that? You mean, instead of passing `Type`?

File: README.md (hunk 1 above)
Comment (Woyten) [reply README.md:28]: Looking into the other crates on crates.io, I could not find any Copyright information in the Readme. In my understanding this information belongs to the LICENSE file.

File: src/lib.rs (hunk 2 above)
Comment (torfmaster) [reply src/lib.rs]: We have multiple occurences of the literal "error_type" and I think there might be a way to put this into a constant. But at a first sight this seems to be not straight forward.

Comment (Woyten) [reply src/lib.rs]: Okay. Did something.

Comment (torfmaster) [reply src/lib.rs]: Great.

**Pull request merged**